rad-cli/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+        println!("{} {}", style("✔").green(), success);
+    }
+
+    pub fn spinner(message: &str) -> ProgressBar {
+        let spinner = ProgressBar::new_spinner();
+        spinner.enable_steady_tick(120);
+        spinner.set_message(message.to_string());
+        spinner
+    }
+
+    pub fn pwhash(secret: SecUtf8) -> Pwhash<keys::CachedPrompt> {
+        let prompt = keys::CachedPrompt::new(secret);
+        Pwhash::new(prompt, KdfParams::recommended())
+    }
+
+    pub fn text_input(message: &str, default: Option<String>) -> String {
+        match default {
+            Some(default) => Input::with_theme(&ColorfulTheme::default())
+                .with_prompt(message)
+                .default(default.into())
+                .interact_text()
+                .unwrap(),
+            None => Input::with_theme(&ColorfulTheme::default())
+                .with_prompt(message)
+                .interact_text()
+                .unwrap(),
+        }
+    }
+
+    pub fn secret_input() -> SecUtf8 {
+        SecUtf8::from(
+            Password::with_theme(&ColorfulTheme::default())
+                .with_prompt("Password")
+                .with_confirmation("Repeat password", "Error: the passwords don't match.")
+                .interact()
+                .unwrap(),
+        )
+    }
+
+    pub mod format {
+
+        use librad::{
+            git::{identities::Project, Urn},
+            profile::Profile,
+        };
+
+        use dialoguer::console::style;
+
+        pub fn profile_list(profiles: &Vec<Profile>, active: &Profile) {
+            for p in profiles {
+                if p.id() == active.id() {
+                    println!(
+                        "  {} {} {}",
+                        style("⊙").magenta(),
+                        &p.id().to_string(),
+                        style("(active)").magenta()
+                    );
+                } else {
+                    println!("  {} {}", style("⋅").white(), &p.id().to_string());
+                }
+            }
+            println!();
+        }
+
+        pub fn project_list(projects: &Vec<Project>) {
+            for p in projects {
+                println!("  {} {}", style("⋅").white(), &p.urn().to_string());
+            }
+            println!("");
+        }
+
+        pub fn seed_config(seed: &str, profile: &Profile, urn: &Urn) {
+            println!("  ⋅ {} {}", style("(Seed)").magenta(), seed);
+            println!(
+                "  ⋅ {} {}",
+                style("(Profile)").magenta(),
+                &profile.id().to_string()
+            );
+            println!("  ⋅ {} {}", style("(Identity)").magenta(), &urn.to_string());
+            println!();
+        }
+
+        pub fn error_detail(detail: &str) {
+            println!("  {} {}", style("⊙").red(), &detail);
+        }
+    }
+}
+
+pub mod id {
+    use librad::git::Urn;
+    pub fn from_urn(urn: &Urn) -> String {
+        urn.to_string().split(":").last().unwrap().to_string()
```

Comment:
`Urn::encode_id` is what you want I think.

Reply:
Great! Thanks.